lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2380,8 +2384,17 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
     for (unsigned i : indices(parsedWrtParams)) {
       auto paramLoc = parsedWrtParams[i].getLoc();
       switch (parsedWrtParams[i].getKind()) {
-      case ParsedAutoDiffParameter::Kind::Index: {
-        unsigned index = parsedWrtParams[i].getIndex();
+      case ParsedAutoDiffParameter::Kind::Named: {
+        auto nameIter = std::find(originalParamNames.begin(),
+                                  originalParamNames.end(),
+                                  parsedWrtParams[i].getName());
+        if (nameIter == originalParamNames.end()) {
+          TC.diagnose(paramLoc,
+                      diag::differentiable_attr_wrt_index_out_of_bounds);
+          return;
+        }
+        // Parameter indices should be in increasing order.
+        unsigned index = std::distance(originalParamNames.begin(), nameIter);
         if ((int)index <= lastIndex) {
           TC.diagnose(paramLoc,
                       diag::differentiable_attr_wrt_indices_must_be_ascending);
```

Comment:
This "must be ascending" diagnostic no longer makes sense now that there are no indices. Some rewording would be nice. Similarly, the check below for "index out of bounds" also doesn't make sense.

Reply:
Done in https://github.com/apple/swift/pull/22000/commits/aec0072e4eb43e7c59c6b73d980030413deaceea.